### ISSUE TITLE
ClassCastException on selecting join field from table with objects

### ISSUE THREAD

**Issue by hlcianfagna** [crate/crate#16555]
### CrateDB version

5.8.1 and 5.8.2

### CrateDB setup information

Tested both with a single node and with 3 nodes

### Problem description

The query below results on a `ClassCastException` if `b.b_id` is selected.
The query still fails even if things are wrapped in a CTE and a WHERE` clause is used to filter out the record where the `try_cast` gives a null value.

### Steps to Reproduce

```sql
CREATE TABLE test.tbla (
	"id" TEXT PRIMARY KEY
	,"document" OBJECT (DYNAMIC) AS (
		"t_id" TEXT
		,"id" TEXT
		)
	,anotherobj OBJECT (DYNAMIC) AS ("id" TEXT)
	);
	
CREATE TABLE test.tblb ("b_id" INTEGER NOT NULL);

INSERT INTO test.tblb
VALUES (1);

INSERT INTO test.tbla
VALUES (
	'a1'
	,{ "t_id" = 'abc'
	,"id" = 'def' }
	,{ "id" = 'd1' }
	);

INSERT INTO test.tbla
VALUES (
	'a2'
	,{ "t_id" = '1'
	,"id" = 'xyz' }
	,{ "id" = 'd2' }
	);

REFRESH TABLE test.tbla;
REFRESH TABLE test.tblb;

SELECT a.*
	,b.b_id
FROM (
	SELECT *
		,try_cast(document ['t_id'] AS INTEGER) AS bId
		,document ['id'] AS id
	FROM test.tbla limit 5
	) AS a
INNER JOIN test.tblb AS b ON b.b_id = a.bId;
```

### Actual Result

```
java.lang.ClassCastException: class java.lang.Integer cannot be cast to class java.lang.String (java.lang.Integer and java.lang.String are in module java.base of loader 'bootstrap')
	at io.crate.types.StringType.valueBytes(StringType.java:69)
	at io.crate.breaker.CellsSizeEstimator$1.estimateSize(CellsSizeEstimator.java:53)
	at io.crate.breaker.CellsSizeEstimator.estimateSize(CellsSizeEstimator.java:83)
	at io.crate.breaker.TypedRowAccounting.accountForAndMaybeBreak(TypedRowAccounting.java:74)
	at io.crate.breaker.TypedRowAccounting.accountForAndMaybeBreak(TypedRowAccounting.java:36)
	at io.crate.rest.action.RestResultSetReceiver.setNextRow(RestResultSetReceiver.java:64)
	at io.crate.protocols.postgres.RetryOnFailureResultReceiver.setNextRow(RetryOnFailureResultReceiver.java:67)
	at io.crate.action.sql.RowConsumerToResultReceiver.consumeIt(RowConsumerToResultReceiver.java:77)
	at io.crate.action.sql.RowConsumerToResultReceiver.lambda$consumeIt$1(RowConsumerToResultReceiver.java:101)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2179)
	at io.crate.action.FutureActionListener.onResponse(FutureActionListener.java:35)
	at org.elasticsearch.action.ActionListenerResponseHandler.handleResponse(ActionListenerResponseHandler.java:57)
	at org.elasticsearch.transport.TransportService$5.handleResponse(TransportService.java:515)
	at org.elasticsearch.transport.TransportService$TimeoutResponseHandler.handleResponse(TransportService.java:1013)
	at org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1082)
	at org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:1062)
	at io.crate.execution.support.NodeActionRequestHandler.lambda$messageReceived$0(NodeActionRequestHandler.java:51)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1773)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1570)
```

### Expected Result

```
+----+----------------------------+--------------+-----+----+------+
| id | document                   | anotherobj   | bid | id | b_id |
+----+----------------------------+--------------+-----+----+------+
| a2 | {"id": "xyz", "t_id": "1"} | {"id": "d2"} |   1 | a2 |   1  |
+----+----------------------------+--------------+-----+----+------+
```

**Comment by mfussenegger** [comment]
Looks like there is an issue in the column pruning logic.

In:

```sql
SELECT
    a.*,
    b.b_id
FROM (
    SELECT
        *,
        try_cast (document['t_id'] AS INTEGER) AS bId,
        document['id'] AS id
    FROM
        test.tbla
    LIMIT 5) AS a
    INNER JOIN test.tblb AS b ON b.b_id = a.bId;
```

The `*, ..., document['id'] as id` has two columns called `id` in the output and the pruning logic removes the second one.
A workaround would be to change `document['id'] as id` to something like `document['id'] as doc_id`

**Comment by mfussenegger** [comment]
Thanks for the report, this should be fixed in 5.8.3 with https://github.com/crate/crate/pull/16558
